### PR TITLE
Deny some schemes handled by external protocol handlers

### PR DESCRIPTION
**Pull request by jumde**
fix https://github.com/brave/browser-laptop/issues/15253

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

DM @jumde for PoC

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

## Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/filtering.js [hunk 1]
```diff
@@ -582,6 +586,8 @@ function registerPermissionHandler (session, partition) {
       } else if (isFullscreen && alwaysAllowFullscreen) {
         // Always allow fullscreen if setting is ON
         response.push(true)
+      } else if (isDeniedScheme.indexOf(protocol.toLowerCase()) >= 0) {
+          response.push(false)
```
Comment (diracdeltas) [inline app/filtering.js]: shouldn't this only be denied if openExternal is the permission?

File: app/filtering.js [hunk 2]
```diff
@@ -526,6 +526,10 @@ function registerPermissionHandler (session, partition) {
     const appState = appStore.getState()
     const isBraveOrigin = mainFrameOrigin.startsWith(`chrome-extension://${config.braveExtensionId}/`)
     const isPDFOrigin = mainFrameOrigin.startsWith(`${pdfjsOrigin}/`)
+
+    //From: https://cs.chromium.org/chromium/src/chrome/browser/external_protocol/external_protocol_handler.cc?l=36
+    const isDeniedScheme = ["afp:", "data:", "disk:", "disks:", "hcp:", "ms-help:", "nntp:", "shell:",
```
Comment (diracdeltas) [inline app/filtering.js]: `data` doesn't seem necessary since it is handled internally by Brave

File: app/filtering.js [hunk 3]
```diff
@@ -582,6 +586,8 @@ function registerPermissionHandler (session, partition) {
       } else if (isFullscreen && alwaysAllowFullscreen) {
         // Always allow fullscreen if setting is ON
         response.push(true)
+      } else if (isOpenExternal && isDeniedScheme.indexOf(protocol.toLowerCase()) >= 0) {
+        response.push(false)
```
Comment (diracdeltas) [inline app/filtering.js]: could you move this to the https://github.com/brave/browser-laptop/pull/15260/files#diff-8a9bac13ef6264feb2b6da7c18d86b3cR562 to make the diff smaller

File: app/filtering.js [hunk 4]
```diff
@@ -526,6 +526,10 @@ function registerPermissionHandler (session, partition) {
     const appState = appStore.getState()
     const isBraveOrigin = mainFrameOrigin.startsWith(`chrome-extension://${config.braveExtensionId}/`)
     const isPDFOrigin = mainFrameOrigin.startsWith(`${pdfjsOrigin}/`)
+
+    // From: https://cs.chromium.org/chromium/src/chrome/browser/external_protocol/external_protocol_handler.cc?l=36
+    const isDeniedScheme = ['afp:', 'disk:', 'disks:', 'hcp:', 'ms-help:', 'nntp:', 'shell:',
```
Comment (diracdeltas) [inline app/filtering.js]: should be called `deniedSchemes` since it's not a function

File: app/filtering.js (hunk 3 above)
Comment (diracdeltas) [reply app/filtering.js]: actually this code block wouldn't be reached otherwise since it's an `else if`

**Comment by bsclifton** [comment]
Closing as this is fixed in brave-core

**Pull request closed without merging**